### PR TITLE
feat(firebase_auth): add tenantId support 

### LINKED ISSUE #1584: [firebase_auth] Multi tenancy ( Identity Platform compatibility )
[Identity Platform](https://cloud.google.com/identity-platform), which relies on Firebase Authentication SDKs for most of the operations ( from my understanding FBA is becoming an abstraction over Identity Platform ), provides multi tenancy support, essentially splitting users by `tenantId`, which [must be provided at login time](https://cloud.google.com/identity-platform/docs/multi-tenancy-authentication#sign_in_with_tenants).
Is there any work in progress or ETA on this feature compatibility for the Flutter plugin?

Thank you for your time.

### PR DESCRIPTION
**Pull request by Ehesp**
## Description

Enabled setting/getting of the `tenantId` property.

## Related Issues

Fixes #1584

Discussion: #5561

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by Salakar — COMMENTED** [review]
Native code LGTM

File: packages/firebase_auth/firebase_auth/test/firebase_auth_test.dart [hunk 1]
```diff
@@ -201,6 +201,27 @@ void main() {
...
+                apiKey: 'apiKey',
+                appId: 'appId',
+                messagingSenderId: 'messagingSenderId',
+                projectId: 'projectId'));
+
+        FirebaseAuthPlatform.instance =
+            FakeFirebaseAuthPlatform(tenantId: 'foo');
+        auth = FirebaseAuth.instanceFor(app: app);
+        expect(auth!.tenantId, 'foo');
+        auth!.tenantId = 'bar';
```
Comment (rrousselGit) [inline packages/firebase_auth/firebase_auth/test/firebase_auth_test.dart:219]: `auth` should probably be `late` so avoid those `!`

File: packages/firebase_auth/firebase_auth/test/firebase_auth_test.dart [hunk 2]
```diff
@@ -201,6 +201,27 @@ void main() {
...
+            name: 'tenantIdTest',
+            options: const FirebaseOptions(
+                apiKey: 'apiKey',
+                appId: 'appId',
+                messagingSenderId: 'messagingSenderId',
+                projectId: 'projectId'));
+
+        FirebaseAuthPlatform.instance =
+            FakeFirebaseAuthPlatform(tenantId: 'foo');
+        auth = FirebaseAuth.instanceFor(app: app);
```
Comment (rrousselGit) [inline packages/firebase_auth/firebase_auth/test/firebase_auth_test.dart:217]:
```suggestion
        auth = FirebaseAuth.instanceFor(app: app);

```

Adding a bit of spacing between variable declarations and expectations, for readability

File: packages/firebase_auth/firebase_auth/test/firebase_auth_test.dart (hunk 1 above)
Comment (rrousselGit) [inline packages/firebase_auth/firebase_auth/test/firebase_auth_test.dart:219]:
```suggestion

        auth!.tenantId = 'bar';

```

Same thing, a bit of spacing to separate the expects from mutations

File: packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart [hunk 3]
```diff
@@ -294,6 +294,15 @@ class Auth extends JsObjectWrapper<auth_interop.AuthJsImpl> {
   /// Currently signed-in [User].
   User? get currentUser => User.getInstance(jsObject.currentUser);
 
+  // Returns the current tenantId for the instance.
+  String? get tenantId {
+    return jsObject.tenantId;
+  }
+
+  set tenantId(String? tenantId) {
+    jsObject.tenantId = tenantId;
```
Comment (rrousselGit) [inline packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart:303]: Shouldn't updating the tenant ID call `auth.setTenantId` on native? Like done in `auth`

**Comment by rrousselGit** [comment]
LGTM with minor style changes and a question

**Comment by rrousselGit** [comment]
Actually I don't see tests. Could you add some? :pray:

File: packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart (hunk 3 above)
Comment (Ehesp) [reply packages/firebase_auth/firebase_auth_web/lib/src/interop/auth.dart:303]: Not for this case - this is the interop between Dart & the JS SDK

**Comment by IchordeDionysos** [comment]
What‘s the status here?

**Comment by russellwheatley** [comment]
@googlebot I consent.

**Review by Salakar — APPROVED** [review]

**Pull request merged**